MapboxCoreNavigation/CLLocation.swift

```diff
@@ -114,17 +114,27 @@ extension CLLocation {
             return stepCoordinates
         }
         
-        return nearbyCoordinates
+        return legProgress.nearbyCoordinates
     }
     
+    func projectedDistance(for distanceRemaining: CLLocationDistance, previousSpeeds: [CLLocationSpeed]) -> CLLocationDistance {
+        var defaultDeadReckonDistance = speed * RouteControllerDeadReckoningTimeInterval
+        
+        // If the user is reducing in speed, start to dead reckon less
+        let averageDelta = previousSpeeds.reduce(0, +) / Double(previousSpeeds.count)
+        if averageDelta < 0 {
+            defaultDeadReckonDistance = defaultDeadReckonDistance + averageDelta
+        }
+        return max(defaultDeadReckonDistance, 0)
+    }
     
     /**
      Given a location and a series of coordinates, compute what the course should be for a the location.
      */
-    func interpolatedCourse(along coordinates: [CLLocationCoordinate2D]) -> CLLocationDirection? {
+    func interpolatedCourse(along coordinates: [CLLocationCoordinate2D], alternateCoordinate: CLLocationCoordinate2D? = nil) -> CLLocationDirection? {
```

Comment:
“Alternate coordinate” is too vague. The name should say what the alternative coordinate is used for, or how it relates to other coordinates being passed in.